File: cpp/src/LC1TwoSum.cpp
#include <vector>
#include <unordered_map>
using namespace std;

/*
Junior
Array
Hash Table
 */

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> map;

        for(int i = 0; i < nums.size(); i++) {
            int needed = target - nums[i];

            if(map.count(needed)) {
                return {map[needed], i};
            }

            map[nums[i]] = i;
        }

        return {};
    }
};

File: Java/src/LC1TwoSum.java
import java.util.HashMap;

/*
Junior
Array
Hash Table
 */

public class LC1TwoSum {
    class Solution {
        public int[] twoSum(int[] nums, int target) {
            HashMap<Integer, Integer> m = new HashMap<>();
            for (int i = 0; i < nums.length; i++) {
                int check = target - nums[i];
                if (m.containsKey(check)) {
                    return new int[] {i, m.get(check)};
                }
                m.put(nums[i], i);
            }
            return new int[] {};
        }
    }
}